app/utils/pages.tsx

```diff
@@ -13,61 +10,76 @@ interface IPage {
+    });
+    if(hasContestantData){
+        const scoringSubpage = {
+            name: "Scoring",
+            path: `/${pageInformation.showStatus}/${pageInformation.showNameAndSeason}/scoring`
+        }
+        const leagueStandingSubpage = {
+            name: "League Standing",
+            path: `/${pageInformation.showStatus}/${pageInformation.showNameAndSeason}/league-standing`
+        }
+        subpages.push(scoringSubpage, leagueStandingSubpage);
+    }
+    return subpages;
 }
 
-export function checkForSubpages (filename:string){
-    return fs.existsSync(path.join(process.cwd(), "app", "leagueData", filename));
+function generatePathObj(pageData:PageInformation, hasContestantData:boolean){
+    const subpages:Array<IPage> = generateSubpages(pageData, hasContestantData);
+    if(pageData.showStatus === "active"){
+        pageData.friendlyName = `Current (${pageData.friendlyName})`
+    }
+    return {
+        name: pageData.friendlyName,
+        subpages: subpages
+    }
 }
 
-export function getPages(): ILeagueLink[] {
+export async function getPages(): Promise<ILeagueLink[]> {
     // Based on availability in leagueConfiguration
-    const filepaths = pagesModule.getPathsToMap();
+    const leagueConfigurationKeys = await getLeagueConfigurationKeys();
     const activeLeaguePaths:Array<ILeagueLink> = [];
     const archiveLeaguePaths:Array<ILeagueLink> = [];
-    filepaths.map((file: string) => {
-        // Needed status for url
-        const { leagueStatus } = pagesModule.getLeagueConfigurationData(file);
-        // Parses filename and converts it to url format
-        const pageStrings = transformFilenameToSeasonNameRepo(file);
-        const subpages:Array<IPage> = [];
-        subpages.push({
-            name: "Contestants",
-            path: `/${leagueStatus}/${pageStrings.urlSlug}/contestants`
-        });
-        if(pagesModule.checkForSubpages(file)){
-            const scoringSubpage = {
-                name: "Scoring",
-                path: `/${leagueStatus}/${pageStrings.urlSlug}/scoring`
-            }
-            const leagueStandingSubpage = {
-                name: "League Standing",
-                path: `/${leagueStatus}/${pageStrings.urlSlug}/league-standing`
-            }
-            subpages.push(scoringSubpage, leagueStandingSubpage);
-        }
-        if(leagueStatus === "active"){
-            pageStrings.friendlyName = `Current (${pageStrings.friendlyName})`
-        }
-        //   Path object created
-        const pathObj = {
-            name: pageStrings.friendlyName,
-            subpages: subpages
-        }
-        if(leagueStatus === "active"){
+    for(const leagueConfigurationKey of leagueConfigurationKeys){
+        const pageData: PageInformation = constructPageInformation(leagueConfigurationKey);
+        const showContestantData = await hasContestantData(pageData.contestantDataKey);
+        const pathObj = generatePathObj(pageData, showContestantData);
```

Comment:
nit: let's rename this variable
```suggestion
        const contestantDataExists = await hasContestantData(pageData.contestantDataKey);
        const pathObj = generatePathObj(pageData, contestantDataExists);
```

Reply:
Committed this change since this was outdated